pytket/qir/conversion/conversion.py

```diff
@@ -0,0 +1,360 @@
+        elif isinstance(op, ClassicalExpBox):
+            for reglist, sizes in [
+                (
+                    inputs,
+                    list(
+                        map(
+                            lambda obj: obj.size  # type: ignore
+                            if isinstance(obj, BitRegister)
+                            else None,
+                            op.get_exp().args,
+                        )
+                    ),
+                ),
+                (outputs, [op.get_n_o()]),
+            ]:
+                for in_width in sizes:
+                    if in_width == 0:
+                        raise ValueError(
+                            "ClassicalExpBox op input or output \
+                            registers have empty widths."
+                        )
+                    com_bits = args[:in_width]
+                    args = args[in_width:]
+                    regname = com_bits[0].reg_name
+                    if com_bits != list(self.cregs[regname]):
+                        raise ValueError(
+                            "ClassicalExpBox ops must act on entire registers."
+                        )
+                    reglist.append(regname)
+        elif isinstance(op, SetBitsOp):
+            for reglist, sizes in [
+                (outputs, [op.n_outputs]),
+            ]:
+                for in_width in sizes:
+                    if in_width == 0:
+                        raise ValueError(
+                            "A value is getting assigned to an empty register."
+                        )
+                    com_bits = args[:in_width]
+                    args = args[in_width:]
+                    regname = com_bits[0].reg_name
+                    if com_bits != list(self.cregs[regname]):
+                        raise ValueError("SetBitOp must act on entire registers.")
+                    reglist.append(regname)
+        return inputs, outputs
+
+    def circuit_to_module(self, circuit: Circuit, module: Module) -> Module:
+        """Populate a PyQir module from a pytket circuit."""
+        for command in circuit:
+            op = command.op
+            if isinstance(op, Conditional):
+                assert op.width == 1  # only one conditional bit
+
+                conditional_circuit = self._rebase_op_to_gateset(
+                    op.op, command.args[op.width :]
+                )
+                condition_bit_index = command.args[0].index[0]
+                condition_name = command.args[0].reg_name
+
+                # condition_ssa = module.module.results[condition_bit_index]
+
+                def condition_block() -> None:
+                    """
+                    Populate recursively the module with the contents of the conditional
+                    sub-circuit when the condition is True.
+                    """
+                    if op.value == 1:
+                        self.circuit_to_module(conditional_circuit, module)
```

Comment:
This confuses me.

* `op.value` is the bit value according to which the circuit should be executed at runtime, right? What's the meaning of these two lines?
* What's with the `true = lambda: condition_block()` lines below, when `condition_block()` is a function that returns a `Module`?
* What happens for other values of `op.value`?